### DISCUSSION TITLE
Registry hooks trigger timing

### DISCUSSION THREAD

**Discussion by AllanHodkinson** [npm/feedback#467] [Registry] [answered]
Hi all,
We've been using registry hooks to send notifications when packages are published as private modules within our organization scope.

We're finding the timing of these notifications to be quite variable, sometimes the hook will trigger within 5 mins, other times it can take up several hours.  I can't find any information online to how these are managed in the registry or if there are performance guidelines.  Can anyone offer any advice?

Many thanks,
Allan

**Comment by MylesBorins** [comment]
thanks for reporting @AllanHodkinson . I've opened an issue with engineering

**Comment by AllanHodkinson** [comment]
Hi Myles,
Is there a way of getting any status updates on the npm hook behaviour?  We're still finding the timing rather variable, it was especially slow at the beginning of August.
Many thanks,
Allan

**Comment by MylesBorins** [comment, marked as answer]
It looks like engineering has fixed what was causing the problem. PTAL and let us know if you are still having issues

**Reply by AllanHodkinson** [reply]
Thanks for letting us know.  Current hook triggering time is about 30 mins after a new package version is published - do you think this will improve or is this the expected behaviour?
Many thanks,
Allan

**Reply by MylesBorins** [reply]
I believe around ~30 minutes is expected behavior
